### DISCUSSION TITLE
Inserting values from a multi-select picker to another table

### DISCUSSION THREAD

**Discussion by ChrisChapman** [Budibase/budibase#10449] [Help] [answered]
I'm having an issue getting something to work which I had assumed would be fairly simple to do.

I have a form which includes a multi-select picker containing values from another table.

When the form is submitted I want to write to **Table A** but have the values selected in the multi-select picker written to **Table B**.

I have set up relationships between the two tables in BB but it only ever writes the first selected value in the picker

My second idea was to use automations to take the input array from the multi-select, use a loop and execute a insert into **Table B** however I cannot get this to work at all.

Can anyone assist with either method A or B ?

Edit - this is a MySQL database

Thanks

**Comment by mmalessa80** [comment, marked as answer]
Which form component are you using? The Multi-select Picker or the Relationship Picker? One-to-Many or Many-to-Many relationship? Your question might suggest that your are using the Multi-select Picker but it is not clear if your choices in that picker are existing rows from Table B and then what is being written to Table B?

![image](https://user-images.githubusercontent.com/106758851/235188939-62b05bc0-5142-43ce-80c4-5b247450258c.png)

**Reply by melohagan** [reply]
Thanks for pointing that out @mmalessa80 ! It's important when dealing with relationship fields to use the relationship picker and not the multi-select.

Here's a couple of docs pages that may help as well @ChrisChapman 👍 
 - https://docs.budibase.com/docs/form-steps#data
 - https://docs.budibase.com/docs/chaining-actions#example-creating-and-linking-two-table-entries-in-one-form

**Reply by ChrisChapman** [reply]
@mmalessa80 @melohagan  I'm using the the multi-select picker on the form as I want to display checkboxes rather than a multi-select dropdown

To give a clear example of the table set up:

![image](https://user-images.githubusercontent.com/1804289/235195181-efbf170a-47bb-42c8-83e9-264384a9e99f.png)


An entry is written to **Table A** on form submit.

The choices in the picker are existing from **Table C** in the database, there is a 1-Many relationship between **Table A** and **Table B**

For each selected item in the picker, I want to insert into **Table B** linking the item between **Table A** and **Table C**

Thanks

**Reply by melohagan** [reply]
Have you had a look at the linked docs pages? If you really want to use a multi-select, have you tried using the `Add column` to set the relationship values?

If you're still struggling send through some screenshots

**Reply by ChrisChapman** [reply]
@melohagan Thanks, I've followed the docs linked and it works using the Relationship Picker.

It would be useful if we could pick between a checkbox/select list in the relationship picker

Thanks again

**Reply by melohagan** [reply]
Awesome, that's great. In future you'll be able to search within the relationship picker which should help use-ability.

I'm not certain checkboxes is a good idea because in theory there could be a lot of relationships, and thus a lot of checkboxes on your page!
